### PR TITLE
feat(dart_frog_cli): added progress indicator for hot reload

### PR DESCRIPTION
**Pull request by dhruvpatidar359**
fixes : #894 

## Status

**READY**

## Description

Added progress indicator during hot reload

## Type of Change

- [x ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart [hunk 1]
```diff
@@ -171,19 +171,25 @@ class DevServerRunner {
...
-      log = logger.info;
-    } else {
-      log = logger.detail;
+  //  keep the params as it is for the progress indicator
+  Future<void> _reload([bool verbose = false, bool progress = true]) async {
+    final logInfoOrDetail = verbose ? logger.info : logger.detail;
+    final logProgress =
+        progress ? logger.progress('[codegen] reloading...') : null;
+
+    if (logProgress == null) {
```
Comment (renancaraujo) [inline packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart:180]:
```suggestion
    if (!showProgress) {
```

File: packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart [hunk 2]
```diff
@@ -171,19 +171,25 @@ class DevServerRunner {
...
   }
 
-  Future<void> _reload([bool verbose = false]) async {
-    final void Function(String) log;
-    if (verbose) {
-      log = logger.info;
-    } else {
-      log = logger.detail;
+  //  keep the params as it is for the progress indicator
+  Future<void> _reload([bool verbose = false, bool progress = true]) async {
```
Comment (renancaraujo) [inline packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart:175]:
```suggestion
  Future<void> _reload([bool verbose = false, bool showProgress = true]) async {
```

Comment (renancaraujo) [inline packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart:175]: At this point we should start using named parameters

File: packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart [hunk 3]
```diff
@@ -171,19 +171,25 @@ class DevServerRunner {
...
+    final logInfoOrDetail = verbose ? logger.info : logger.detail;
```
Comment (renancaraujo) [inline packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart:176]: I don't understand the refactor on the name from `log` to `logInfoOrDetail`.

File: packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart [hunk 4]
```diff
@@ -171,19 +171,25 @@ class DevServerRunner {
...
+      logInfoOrDetail('[codegen] reloading...');
     }
 
-    log('[codegen] reloading...');
     _isReloading = true;
     await _codegen();
     _isReloading = false;
-    log('[codegen] reload complete.');
+
+    if (logProgress == null) {
```
Comment (renancaraujo) [inline packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart:188]: The nullability of `logProgress` is not the condition here, `showProgress` is.

File: packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart [hunk 5]
```diff
@@ -395,9 +401,9 @@ class DevServerRunner {
 
   /// Regenerates the dev server code and sends a hot reload signal to the
   /// server.
-  Future<void> reload() async {
+  Future<void> reload({bool infoOrDetail = false, bool progress = true}) async {
     if (isCompleted || !isServerRunning || _isReloading) return;
-    return _reload(true);
+    return _reload(infoOrDetail, progress);
```
Comment (renancaraujo) [inline packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart:406]: Is this breaking API addressed somewhere? Verbose was true before.

**Review by renancaraujo — CHANGES_REQUESTED** [review]
Apart from the points added to this review, o think we should not use progress on the daemon, it makes the event a little hard to parse.

**Comment by dhruvpatidar359** [comment]
> Apart from the points added to this review, o think we should not use progress on the daemon, it makes the event a little hard to parse.



> Apart from the points added to this review, o think we should not use progress on the daemon, it makes the event a little hard to parse.

Alright that means for showing the time, we can use the timer sort of thing . But we can't use the progress ,i can't show the progress sort of thing . Should i implement kind of custom logging here ?

**Comment by renancaraujo** [comment]
As we are working on https://github.com/VeryGoodOpenSource/dart_frog/issues/1016, the code to be changed here tends to move around a little in the next few days, so I think it is better to wait for that before tackling this. 

Thanks for opening the issue and your time dedicated to this.
